products/green_fast_track/models/intermediate/flags/int_flags__all.sql

```diff
@@ -5,23 +5,43 @@
     ]
 ) }}
 
-SELECT
-    bbl,
-    variable_type,
-    variable_id,
-    NULL::double precision AS distance
-FROM {{ ref('int_flags__zoning') }}
-UNION ALL
-SELECT
-    bbl,
-    variable_type,
-    variable_id,
-    NULL::double precision AS distance
-FROM {{ ref('int_flags__edesignation') }}
-UNION ALL
-SELECT
-    bbl,
-    variable_type,
-    variable_id,
-    distance
-FROM {{ ref('int_flags__spatial') }}
+WITH variables AS (
+    SELECT * FROM {{ ref('variables') }}
+),
+
+all_variable_ids AS (
+    SELECT
+        bbl,
+        variable_type,
+        variable_id,
+        NULL::double precision AS distance
+    FROM {{ ref('int_flags__zoning') }}
+    UNION ALL
+    SELECT
+        bbl,
+        variable_type,
+        variable_id,
+        NULL::double precision AS distance
+    FROM {{ ref('int_flags__edesignation') }}
+    UNION ALL
+    SELECT
+        bbl,
+        variable_type,
+        variable_id,
+        distance
+    FROM {{ ref('int_flags__spatial') }}
+),
+
+all_flags AS (
+    SELECT
+        bbl,
+        variables.flag_id_field_name,
+        all_variable_ids.variable_type,
+        variable_id,
+        distance
+    FROM all_variable_ids LEFT JOIN variables
+        ON all_variable_ids.variable_type = variables.variable_type
+)
+
+SELECT * FROM all_flags
+ORDER BY bbl ASC, flag_id_field_name ASC, variable_type ASC
```

Comment:
`int_flags__all` now has a new column called `flag_id_field_name`

this column replaces the downstream use of `variable_type` because, when pivoting from a long to wide table, we now care about which flag (aka survery question) each `variable_id` like "Historic Lamppost" is related to rather than the variable (aka source data) it came from